bson/bsoncodec/string_codec.go

```diff
@@ -72,6 +72,12 @@ func (sc *StringCodec) DecodeValue(dctx DecodeContext, vr bsonrw.ValueReader, va
 		if err != nil {
 			return err
 		}
+	case bsontype.Binary:
+		bin, _, err := vr.ReadBinary()
```

Comment:
I think we may actually want to care about the subtype here. For instance, the UUID subtypes should have their string versions look like the UUID that generated them.

Reply:
this code was removed